app/llm/performance_enhanced.py

```
@@ -0,0 +1,142 @@
+                if key in self._cache:
+                    self._hits += 1
+                    self._cache.move_to_end(key)
+                    return self._cache[key]
+                else:
+                    self._misses += 1
+                    return None
+            except Exception as e:
+                logger.error(f"Cache retrieval error: {e}")
+                return None
+
+    def set(self, key: str, value: Any) -> bool:
```

Ruff (0.8.2)

Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)